convert/convert_test.go

```diff
@@ -0,0 +1,264 @@
+			got, err := validConversion(tt.args.from, tt.args.to)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("validConversion() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("validConversion() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestParseFormat(t *testing.T) {
+	type args struct {
+		key string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    Format
+		wantErr bool
+	}{
+		{
+			name: "parses valid values",
+			args: args{
+				key: "kong-gateway",
+			},
+			want:    FormatKongGateway,
+			wantErr: false,
+		},
+		{
+			name: "parses values in a case-insensitive manner",
+			args: args{
+				key: "koNNect",
+			},
+			want:    FormatKonnect,
+			wantErr: false,
+		},
+		{
+			name: "parse fails with invalid values",
+			args: args{
+				key: "k42",
+			},
+			want:    "",
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := ParseFormat(tt.args.key)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ParseFormat() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("ParseFormat() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_kongServiceToKonnectServicePackage(t *testing.T) {
+	type args struct {
+		service file.FService
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    file.FServicePackage
+		wantErr bool
+	}{
+		{
+			name: "converts a kong service to service package",
+			args: args{
+				service: file.FService{
+					Service: kong.Service{
+						Name: kong.String("foo"),
+						Host: kong.String("foo.example.com"),
+					},
+				},
+			},
+			want: file.FServicePackage{
+				Name:        kong.String("foo"),
+				Description: kong.String("placeholder description for foo service package"),
+				Versions: []file.FServiceVersion{
+					{
+						Version: kong.String("v1"),
+						Implementation: &file.Implementation{
+							Type: utils.ImplementationTypeKongGateway,
+							Kong: &file.Kong{
+								Service: &file.FService{
+									Service: kong.Service{
+										Host: kong.String("foo.example.com"),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "converts fails for kong services without a name",
+			args: args{
+				service: file.FService{
+					Service: kong.Service{
+						ID:   kong.String("service-id"),
+						Host: kong.String("foo.example.com"),
+					},
+				},
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := kongServiceToKonnectServicePackage(tt.args.service)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("kongServiceToKonnectServicePackage() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			zeroOutID(&got)
+			if !reflect.DeepEqual(got, tt.want) {
+				assert.Equal(t, tt.want, got)
+			}
+		})
+	}
+}
+
+func zeroOutID(sp *file.FServicePackage) {
```

Comment:
Same comment as above about returning a copy instead of editing in place.

I know that this function is in a test (which, one might argue, limits the blast), but I don't feel that tests should have a lower quality bar than tested code.

Also, the rightmost part of the `a && b && c && d` is superfluous: if `v.Implementation.Kong.Service.ID` is nil, then setting it to nil won't hurt